images/converter/default.go

```diff
@@ -349,22 +349,23 @@ func clearDockerV1DummyID(cfg DualConfig) (bool, error) {
 }
 
 // ObjectWithMediaType represents an object with a MediaType field
+// Deprecated
 type ObjectWithMediaType struct {
 	// MediaType appears on Docker manifests and manifest lists.
 	// MediaType does not appear on OCI manifests and index
 	MediaType string `json:"mediaType,omitempty"`
 }
 
 // DualManifest covers Docker manifest and OCI manifest
+// Deprecated: use github.com/opencontainers/image-spec/specs-go/v1.Manifest
 type DualManifest struct {
 	ocispec.Manifest
-	ObjectWithMediaType
 }
 
 // DualIndex covers Docker manifest list and OCI index
+// Deprecated: use github.com/opencontainers/image-spec/specs-go/v1.Index
 type DualIndex struct {
 	ocispec.Index
-	ObjectWithMediaType
 }
```

Comment:
Heads up; i had to make changes here, because the existing type's `MediaType` conflicted with the re-added `MediaType` in the image-spec; see https://github.com/containerd/containerd/pull/6263#issuecomment-972330380

```
# github.com/containerd/containerd/images/converter
images/converter/default.go:162:33: ambiguous selector manifest.MediaType
images/converter/default.go:163:11: ambiguous selector manifest.MediaType
images/converter/default.go:244:30: ambiguous selector index.MediaType
images/converter/default.go:245:8: ambiguous selector index.MediaType
make: *** [Makefile:227: bin/ctr] Error 2
```

consumers of the containerd modules may run into this if they update the image-spec module